helix-core/src/main/java/org/apache/helix/HelixManager.java

```diff
@@ -425,6 +426,19 @@ void addExternalViewChangeListener(org.apache.helix.ExternalViewChangeListener l
    */
   boolean isLeader();
 
+  /**
+   * Checks if the cluster manager is leader and sets its ZK session in
+   * {@link ControllerLeaderSession}.
+   *
+   * @param controllerLeaderSession To include ZK session of the cluster manager in return
+   *
+   * @return true if this is a controller and a leader of the cluster. Zk session of the cluster
+   * manager is set in controllerLeaderSession
+   */
+  default boolean isLeader(ControllerLeaderSession controllerLeaderSession) {
```

Comment:
This could be confusion because there is no "leader" concept for a Helix manager. Leader only applies to a controller instance, but HelixManager is a broader concept here.